packages/salesforcedx-sobjects-faux-generator/test/integration/sObjectDescribe.test.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import rimraf = require('rimraf');
+import { expect } from 'chai';
+import * as path from 'path';
+import {
+  SObjectCategory,
+  SObjectDescribe
+} from '../../src/describe/sObjectDescribe';
+import * as util from './integrationTestUtil';
+
+const PROJECT_NAME = `project_${new Date().getTime()}`;
+const CUSTOM_OBJECT_NAME = 'MyCustomObject__c';
+const CUSTOM_FIELD_FULLNAME = CUSTOM_OBJECT_NAME + '.MyCustomField__c';
+const SIMPLE_OBJECT_DIR = path.join(
+  'test',
+  'integration',
+  'config',
+  'simpleObjectAndField',
+  'objects'
+);
+
+//const util = new TestUtil();
+const sobjectdescribe = new SObjectDescribe();
+
+// tslint:disable:no-unused-expression
+describe('Fetch sObjects', function() {
+  // tslint:disable-next-line:no-invalid-this
+  this.timeout(40000);
+  let username: string;
+
+  before(async function() {
+    const sourceFolder = path.join(
+      __dirname,
+      '..',
+      '..',
+      '..',
+      SIMPLE_OBJECT_DIR
+    );
+    const permSetName = 'AllowRead';
+    await util.createSFDXProject(PROJECT_NAME);
+    username = await util.createScratchOrg(PROJECT_NAME);
+
+    await util.push(sourceFolder, PROJECT_NAME, username);
+    const permissionSetId = await util.createPermissionSet(
+      permSetName,
+      username
+    );
+    await util.createFieldPermissions(
+      permissionSetId,
+      CUSTOM_OBJECT_NAME,
+      CUSTOM_FIELD_FULLNAME,
+      username
+    );
+    await util.assignPermissionSet(permSetName, username);
+  });
+
+  after(function() {
+    const projectPath = path.join(process.cwd(), PROJECT_NAME);
+    rimraf.sync(projectPath);
+  });
+
+  it('Should be able to call describeGlobal', async function() {
+    const cmdOutput = await sobjectdescribe.describeGlobal(
+      process.cwd(),
+      SObjectCategory.CUSTOM,
+      username
+    );
+    expect(cmdOutput.length).to.be.equal(1);
+    expect(cmdOutput[0]).to.be.equal(CUSTOM_OBJECT_NAME);
+  });
+
+  it('Should be able to call describeSObject on custom object', async function() {
+    const cmdOutput = await sobjectdescribe.describeSObject(
+      process.cwd(),
+      CUSTOM_OBJECT_NAME,
+      username
+    );
+    expect(cmdOutput.name).to.be.equal(CUSTOM_OBJECT_NAME);
+    expect(cmdOutput.custom).to.be.true;
+    expect(cmdOutput.fields.length).to.be.greaterThan(0);
+    const customField = cmdOutput.fields[cmdOutput.fields.length - 1];
+    expect(customField.custom).to.be.true;
+    expect(customField.precision).to.be.equal(18);
+    expect(customField.scale).to.be.equal(0);
+    expect(customField.name).to.be.equal('MyCustomField__c');
+  });
+
+  it('Should be able to call describeSObject on standard object', async function() {
+    // const cmdOutput = await sobjectdescribe.describeSObject(
```

Comment:
The command output needs a buffer for standard objects. The amount of json is too large and gets truncated. I'll do this as part of adding explicit refresh command.